lua/treesitter-context/context.lua

```diff
@@ -45,10 +45,29 @@ local function get_parent_nodes(langtree, range)
   return ret
 end
 
+--- @param winid integer
+--- @param percent string
+--- @return integer
+local function max_lines_from_string(winid, percent)
+  local win_height = api.nvim_win_get_height(winid)
+  local percent_s = string.match(config.max_lines, '^(%d+)%%$')
```

Comment:
Generally prefer the method form since it's a bit more concise.

```suggestion
  local percent_s = config.max_lines:match('^(%d+)%%$')
```